save-backend/src/main/kotlin/com/saveourtool/save/backend/service/OrganizationService.kt

```diff
@@ -21,11 +22,15 @@ class OrganizationService(
      * @param organization an [Organization] to store
      * @return organization's id, should never return null
      */
-    @Suppress("UnsafeCallOnNullableType")
-    fun getOrSaveOrganization(organization: Organization): Pair<Long, OrganizationSaveStatus> {
-        val (organizationId, organizationSaveStatus) = organizationRepository.findByName(organization.name)?.let {
-            Pair(it.id, OrganizationSaveStatus.EXIST)
-        } ?: Pair(organizationRepository.save(organization).id, OrganizationSaveStatus.NEW)
+    @Suppress("UnsafeCallOnNullableType", "TooGenericExceptionCaught")
+    @Transactional
+    fun saveOrganization(organization: Organization): Pair<Long, OrganizationSaveStatus> {
+        val (organizationId, organizationSaveStatus) = if (organizationRepository.validateOrganizationName(organization.name) != 0L) {
+            organizationRepository.saveOrganizationName(organization.name)
+            Pair(organizationRepository.save(organization).id, OrganizationSaveStatus.NEW)
+        } else {
+            Pair(0L, OrganizationSaveStatus.CONFLICT)
```

Comment:
what's about case then organization exists already?

Reply:
If organization with the same name already exists, we get CONFLICT

Reply:
Ant this method is not planned for updating an organization